src/diffusers/pipelines/free_noise_utils.py

```diff
@@ -30,6 +34,100 @@
+    def __init__(
+        self,
+        module: nn.Module,
+        split_size: int = 1,
+        split_dim: int = 0,
+        input_kwargs_to_split: List[str] = ["hidden_states"],
+    ) -> None:
+        super().__init__()
+
+        self.module = module
+        self.split_size = split_size
+        self.split_dim = split_dim
+        self.input_kwargs_to_split = set(input_kwargs_to_split)
+
+    def forward(self, *args, **kwargs) -> Union[torch.Tensor, Tuple[torch.Tensor]]:
+        r"""Forward method for the `SplitInferenceModule`.
+
+        This method processes the input by splitting specified keyword arguments along a given dimension, running the
+        underlying module on each split, and then concatenating the results. The splitting is controlled by the
+        `split_size` and `split_dim` parameters specified during initialization.
+
+        Args:
+            *args (`Any`):
+                Positional arguments that are passed directly to the `module` without modification.
+            **kwargs (`Dict[str, torch.Tensor]`):
+                Keyword arguments passed to the underlying `module`. Only keyword arguments whose names match the
+                entries in `input_kwargs_to_split` and are of type `torch.Tensor` will be split. The remaining keyword
+                arguments are passed unchanged.
+
+        Returns:
+            `Union[torch.Tensor, Tuple[torch.Tensor]]`:
+                The outputs obtained from `SplitInferenceModule` are the same as if the underlying module was inferred
+                without it.
+                - If the underlying module returns a single tensor, the result will be a single concatenated tensor
+                along the same `split_dim` after processing all splits.
+                - If the underlying module returns a tuple of tensors, each element of the tuple will be concatenated
+                along the `split_dim` across all splits, and the final result will be a tuple of concatenated tensors.
+
+        Workflow:
+            1. The keyword arguments specified in `input_kwargs_to_split` are split into smaller chunks using
+            `torch.split()` along the dimension `split_dim` and with a chunk size of `split_size`.
+            2. The `module` is invoked once for each split with both the split inputs and any unchanged arguments
+            that were passed.
+            3. The output tensors from each split are concatenated back together along `split_dim` before returning.
+
+        Example:
+            ```python
+            >>> import torch
+
+            >>> model = nn.Linear(1000, 1000)
+            >>> split_module = SplitInferenceModule(
+            ...     model, split_size=2, split_dim=0, input_kwargs_to_split=["input_data"]
+            ... )
+
+            >>> input_tensor = torch.randn(42, 1000)
+            >>> # Will split the tensor into 21 slices of shape [2, 1000].
+            >>> output = split_module(input_data=input_tensor)
```

Comment:
```suggestion
            >>> output = split_module(input=input_tensor)
```

My bad here, nn.Linear takes `input` here and not `input_data`